### PR TITLE
[dv,full_chip,pwrmgr] Add pwrmgr_sleep_power_glitch_reset test

### PR DESCRIPTION
**Pull request by matutem**
Signed-off-by: Guillermo Maturana <maturana@google.com>

### CODE REVIEW

File: sw/device/tests/sim_dv/meson.build [hunk 1]
```diff
@@ -176,6 +174,26 @@ sw_tests += {
   }
 }
```
Comment (mcy) [inline sw/device/tests/sim_dv/meson.build:176]: Please remember to have a Bazel rule for this, too.

File: sw/device/tests/sim_dv/pwrmgr_sleep_power_glitch_test.c [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+    // Configure deep sleep.
+    pwrmgr_testutils_enable_low_power(&pwrmgr,
+                                      kDifPwrmgrWakeupRequestSourceFive, 0);
+
+    // This causes core_sleeping to rise and triggers the injection of the
+    // power glitch. Enter low power mode.
+    LOG_INFO("Issue WFI to enter sleep");
+    wait_for_interrupt();
+
```
Comment (mcy) [inline sw/device/tests/sim_dv/pwrmgr_sleep_power_glitch_test.c:54]: Drop this newline.

**Review by drewmacrae — APPROVED** [review]

**Comment by drewmacrae** [comment]
The Bazel rules LGTM

**Review by tjaychen — APPROVED** [review]
this looks good to me.
Just one item from @mcy to address.

**Pull request merged**